### PR TITLE
Fix MnemonicToByteArray

### PR DESCRIPTION
**Pull request by bartekn**
All test vectors from BIP-0039 pass.

Fix https://github.com/tyler-smith/go-bip39/issues/3

### CODE REVIEW

**Comment by cornfeedhobo** [comment]
@tyler-smith Could you give this a look?

File: bip39_test.go [hunk 1]
```diff
@@ -14,23 +14,18 @@ type Vector struct {
...
-		// Test Vectors 0, 4, and 8 do not work as intended.
-		if (i != 0) && (i != 4) && (i != 8) {
-			assert.Nil(t, err)
-		} else {
-			assert.NotNil(t, err)
-		}
-		seed := NewSeed(mnemonic, "TREZOR")
+		seed, err := NewSeedWithErrorChecking(mnemonic, "TREZOR")
+		assert.Equal(t, vector.seed, hex.EncodeToString(seed))
+		assert.NoError(t, err)
```
Comment (cornfeedhobo) [inline bip39_test.go]: Wouldn't you want to assert this before the previous `Equal`?

**Comment by bartekn** [comment]
@cornfeedhobo thanks!

**Comment by tyler-smith** [comment]
Thanks for the ping, this had fallen off my radar.

Thanks for the fix @bartekn

**Comment by bartekn** [comment]
@tyler-smith yeah, no problem. You can also check https://github.com/tyler-smith/go-bip39/pull/6 if you want.

**Pull request merged**